chainerx_cc/chainerx/cuda/cuda_device/dot.cu

```diff
@@ -140,7 +140,7 @@ public:
                     &zero,
                     internal::GetRawOffsetData(out_contiguous),
                     data_type,
-                    n,
+                    std::max(int64_t{1}, n),
```

Comment:
Could you just leave a comment why this change was necessary?

Reply:
During backward computation of solve for zero-sized input, I was getting an error about incorrect argument to `GemmEx`. Docs say that this argument should not be less than one, it also says that for `lda`, `ldb`, but tests did not complain about them.

Reply:
Could you check the actual shapes of the inputs `a` and `b` to `Dot`?

Reply:
```
test_linalg.py::test_Solve_param_0_{b_columns=(), dtypes=('float32', 'float32'), shape=(0, 0)}[cuda:0]
GemmEx Call! N 0  M  0  K  0  ldb  1  lda  1
```
```
test_linalg.py::test_SVD_param_1_{compute_uv=True, dtype='float32', full_matrices=False, shape=(0, 3)}[cuda:0]
GemmEx Call! N 0  M  0  K  0  ldb  3  lda  3
```
where `ldb == b_cast_layout.ld`, `lda == a_cast_layout.ld`